dimitriosv/week2/index.html

```diff
@@ -0,0 +1,316 @@
+            this.setState({
+              inputListName: evt.target.value
+          });
+        },
+        render: function() {
+            return  <div className="container">
+                        <span>Add new list with name: </span>
+                        <input onChange={this.handleChange} type="text"/>
+                        <button type="button" onClick={this.createList}>Add</button>
+                        <div className="llistcont">
+                            {this.state.lists.map(this.eachList)}
+                        </div>
+                    </div>
+        },
+    });
+
+    var List = React.createClass({
+        getInitialState: function() {
+            return {
+                    inputMessageName:"",
+                    listId:this.props.listId
+                };
+        },
+        render: function() {
+            return  <div className="llist">
+                        <div className="listname" >{this.props.listName}</div>
+                        <div>
+                            <input onChange={this.handleMessageChange} type="text"/>
+                            <button type="button" onClick={this.addMessageChild}>Submit</button>
+                        </div>
+                         {/*console.log(thats how you comment inside return)*/}
+                         <Messages  deleteMessageParent={this.props.deleteMessageParent}
+                                    archiveMessageParent={this.props.archiveMessageParent}
+                                    addMessageParent={this.props.addMessageParent}
+                                    messages={this.filterMessages()} 
+                                    allLists={this.props.allLists}/> 
+                    </div>
+        },
+        addMessageChild: function() {
+            if (this.state.inputMessageName.length==0 || this.state.inputMessageName.length>200) {
+                alert("Message length not valid!")
+            } else {
+                this.props.addMessageParent(this.state.listId, this.state.inputMessageName)
+            }
+        },
+        filterMessages: function() {
+            var filteredArray=[]
+            for (index = 0; index < this.props.allMessages.length; ++index) {
+                if (this.props.allMessages[index].belongsToList==this.props.listId)
+                    {
+                        filteredArray.push(this.props.allMessages[index])
+                    }
+            }
+           return filteredArray;
+        },
+        handleMessageChange: function(evt) {
+            this.setState({
+              inputMessageName: evt.target.value
+          });
+        }
+    })
+
+
+    var Messages = React.createClass({
+        render: function () {
+            var deleteMessageParent=this.props.deleteMessageParent
+            var archiveMessageParent=this.props.archiveMessageParent
+            var addMessageParent=this.props.addMessageParent
+            var allLists=this.props.allLists
```

Comment:
instead of keeping an extra refference to your functions here, you could `.bind` to the map function. Like so:

```javascript
this.props.messages.map(function(message,i) {
   // logic here
}.bind(this))
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind